src/main/java/sirius/biz/jobs/batch/file/XMLValidator.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.jobs.batch.file;
+
+import org.xml.sax.ErrorHandler;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+import sirius.biz.process.ProcessContext;
+import sirius.biz.process.logs.ProcessLog;
+import sirius.kernel.commons.Monoflop;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.health.Exceptions;
+import sirius.web.security.UserContext;
+
+import javax.xml.XMLConstants;
+import javax.xml.transform.Source;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * Validates an XML source against an XSD source and keeps track if any error occurred during validation.
+ */
+public class XMLValidator {
+
+    private final ProcessContext process;
+
+    /**
+     * Creates a validator that logs errors into the given process context.
+     *
+     * @param process the context to log errors into
+     */
+    public XMLValidator(ProcessContext process) {
+        this.process = process;
+    }
+
+    /**
+     * Determines if the  {@link Source xml source} is valid by validating it with the given {@link Source xsd source}.
+     *
+     * @param xml the XML source to be validated
+     * @param xsd the XSD source used as the schema for the validator
+     * @return <tt>true</tt> if the xml source is valid, <tt>false</tt> otherwise
+     * @throws SAXException in case of an invalid xsd source
+     */
+    public boolean validate(Source xml, Source xsd) throws SAXException {
+        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        Schema schema = schemaFactory.newSchema(xsd);
+
+        Validator validator = schema.newValidator();
+        String language = UserContext.getCurrentUser().getLang();
+        // Apply a workaround for using the english translations as those are not stored in an extra "en" resource
+        // bundle but in the root one instead.
+        Locale locale = Strings.areEqual(language, "en") ? Locale.ROOT : new Locale(language);
+        validator.setProperty("http://apache.org/xml/properties/locale", locale);
+
+        XMLValidatorErrorHandler errorHandler = new XMLValidatorErrorHandler(process);
+        validator.setErrorHandler(errorHandler);
+
+        try {
+            validator.validate(xml);
+        } catch (SAXParseException | IOException e) {
+            // Cause of the exception is already logged via validator's error handler.
+            Exceptions.ignore(e);
```

Comment:
Error handler will only handle SAXParseException. So we will swallow IOExceptions here.